hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/processor/ProcessorSupplierWithService.java

```diff
@@ -50,7 +52,12 @@ private ProcessorSupplierWithService(
 
     @Override
     public void init(@Nonnull ProcessorSupplier.Context context) {
-        this.serviceContext = serviceFactory.createContextFn().apply(context);
+        HazelcastInstanceImpl hazelcastInstance = (HazelcastInstanceImpl) context.jetInstance().getHazelcastInstance();
+        ManagedContext managedContext = hazelcastInstance.getSerializationService().getManagedContext();
+        serviceContext = serviceFactory.createContextFn().apply(context);
+        if (managedContext != null) {
```

Comment:
is this allowed to ever return null? I would expect a properly constructed instance to always return `HazelcastManagedContext`

Reply:
I am thinking whether we should also apply the context to the function itself. 

it sound strange at first. but it allows you to provide e.g. `SpringServiceFactory` which will give you an arbitrary Spring bean.